File: frontend/attendance_app/lib/shared/models/qr_token.dart
class QrToken {
  const QrToken({
    required this.id,
    required this.shiftId,
    required this.token,
    required this.expiresAt,
    required this.isUsed,
  });

  final String id;
  final String shiftId;
  final String token;
  final DateTime expiresAt;
  final bool isUsed;

  factory QrToken.fromJson(Map<String, dynamic> json) => QrToken(
        id: json['id'] as String,
        shiftId: json['shiftId'] as String,
        token: json['token'] as String,
        expiresAt: DateTime.parse(json['expiresAt'] as String).toLocal(),
        isUsed: json['isUsed'] as bool? ?? false,
      );
}

File: backend/AttendanceApi/AttendanceApi.Domain/Entities/QrToken.cs
namespace AttendanceApi.Domain.Entities;

public class QrToken
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid ShiftId { get; set; }
    public string Token { get; set; } = string.Empty;
    public DateTime ExpiresAt { get; set; }
    public bool IsUsed { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Shift? Shift { get; set; }
}
